test/ci/verify-templates.test.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+
+// These tests run in a CI context and verify that some important guarantees
+// hold for all the functions in the repository.
+
+if(process.env.CI) {
```

Comment:
What was your thinking around not making these not as part of the standard `npm test` suite?

I feel like in some ways this might be more annoying to a developer to think they've passed all their tests locally, only to fail in the pr due to some mysterious suite of tests

Reply:
My initial understanding of this work was that these would run just for PR validation, probably to avoid slowing down the main test suite too much. That said, on my local system these execute in less than a second; I can definitely get behind adding these to the main test suite (possibly replacing `all-templates.test.js` with this file) if there's no objections to having them there.